src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1558,360 @@ function writeManifestFile(
 
   return store.set('manifest', installedInfo)
 }
+
+export const downloadPatcher = async () => {
+  if (!existsSync(ipdtPatcher)) {
+    try {
+      const { downloadIPDTForOS } = await import('@hyperplay/patcher')
+      await downloadIPDTForOS(toolsPath)
+
+      const version = await getIpdtPatcherVersion()
+      const versionFile = path.join(toolsPath, 'ipdt_version.txt')
+
+      logInfo(
+        `IPDT patcher ${version} downloaded successfully`,
+        LogPrefix.HyperPlay
+      )
+      await writeFile(versionFile, version)
+
+      if (!isWindows) {
+        await chmod(ipdtPatcher, 0o755)
+      }
+    } catch (error) {
+      captureException(error, {
+        extra: {
+          method: 'downloadPatcher'
+        }
+      })
+
+      logError(`Error downloading IPDT: ${error}`, LogPrefix.HyperPlay)
+      throw new Error('Error downloading IPDT')
+    }
+  }
+}
+
+const getIpdtPatcherVersion = async () => {
+  const { stdout } = await spawnAsync(ipdtPatcher, ['-version'])
+  return `${stdout}`.split(' ')[2]
+}
+
+export async function downloadGameIpdtManifest(
+  appName: string,
+  version: string
+) {
+  try {
+    const {
+      install: { channelName, platform },
+      is_installed
+    } = getGameInfo(appName)
+    if (!channelName || !platform || !is_installed) return false
+
+    // download only if the manifest file is not already downloaded and its the same version
+    const manifestName = `${appName}-${platform}-${version}.json`
+    const manifestPath = path.join(ipdtManifestsPath, manifestName)
+    if (existsSync(manifestPath)) return false
+
+    const releaseId = generateID(appName, version)
+    const manifestUrl = await getHyperPlayReleaseManifest(releaseId, platform)
+    if (!manifestUrl) return false
```

Comment:
from what I see, the return value of this function is not being used anywhere. For errors why not throw them instead of silently returning a boolean?

Reply:
yeah, true

Reply:
Changed it now to throw errors and don't return anything else.